lib/spack/spack/spec.py

```diff
@@ -1401,7 +1401,9 @@ def build_hash(self, length=None):
         return self._cached_hash(ht.build_hash, length)
 
     def full_hash(self, length=None):
-        """Hash  to determine when to rebuild packages in the build pipeline.
+        """
+        Hash to determine when to rebuild packages in the build pipeline and
+        to track build failures.
```

Comment:
I would include the uses in the details below and make the top line descriptive -- "hash of build, run, and link dependencies + package.py file"

Reply:
Restoring the original comment since tracking build failures doesn't currently use `full_hash`.

Even so, the description should not include build [dependencies] since `hash_types.full_hash` does not currently include them.